src/components/pages/user/components/delete-account.tsx

```diff
@@ -0,0 +1,156 @@
+import * as React from 'react'
+
+import {useViewer} from 'context/viewer-context'
+import {Dialog, Transition} from '@headlessui/react'
+import {trpc} from 'trpc/trpc.client'
+import toast from 'react-hot-toast'
+import {useRouter} from 'next/router'
+
+const DeleteAccount: React.FC<{}> = () => {
+  const [dialogIsOpen, setDialogIsOpen] = React.useState(false)
+
+  return (
+    <div className="mt-12 space-y-2">
+      <h2 className="pb-3 md:pb-4 text-lg font-medium md:font-normal md:text-xl leading-none text-red-500 ">
+        Danger Zone
+      </h2>
+      <p className="opacity-80">
+        Permanently delete your account, this is not reversable.
+      </p>
+      <button
+        className="inline-block px-4 py-3 text-gray-900 hover:text-white dark:text-white border-red-600 border rounded focus:outline-none hover:bg-red-700"
+        onClick={() => setDialogIsOpen(true)}
+      >
+        Delete Account
+      </button>
+      <AccountDeletionDialog
+        isOpen={dialogIsOpen}
+        setIsOpen={setDialogIsOpen}
+      />
+    </div>
+  )
+}
+
+const AccountDeletionDialog = ({
+  isOpen,
+  setIsOpen,
+}: {
+  isOpen: boolean
+  setIsOpen: Function
+}) => {
+  const {viewer, logout} = useViewer()
+  const router = useRouter()
+
+  const deleteAccount = trpc.user.deleteAccount.useMutation({
+    onSuccess: async (data) => {
+      toast.success(`Account for ${viewer?.email} deleted.`, {
+        duration: 6000,
+        icon: '✅',
+      })
+      localStorage.clear()
+      logout()
+      router.push('/')
```

Comment:
on success,

displays a toast, clears local storage, logs the user out, and navigates back home. 

If we don't log the user out the user will be signed in but their account will be the scrambled email that the `DeleteUserWorker` sets